### ISSUE TITLE
[Bug]: Split Button fails accessibility test

### ISSUE THREAD

**Issue by rvndr29** [microsoft/fluentui#23840]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
npx envinfo --system --npmPackages '{@fluent/*}' --browsers  
npx: installed 1 in 3.041s

  System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz
    Memory: 17.65 GB / 31.78 GB
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Spartan (44.22000.120.0), Chromium (103.0.1264.49)
    Internet Explorer: 11.0.22000.120
```

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/button

## Actual Behavior
Scroll to the Split Button section on the fluent UI Button example page https://developer.microsoft.com/en-us/fluentui#/controls/web/button
Please run automated check using the "Accessibility Insights for Web" plugin
The plugin reports accessibility issue - Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies
## Expected Behavior
The button should not create an accessibility issue
![image](https://user-images.githubusercontent.com/106670862/177995269-a9ab6ad8-ba49-4803-9fef-0efa7b713326.png)
![image](https://user-images.githubusercontent.com/106670862/177995698-0370be60-2045-4975-ab0e-cd24bedddf37.png)

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
@rvndr29 - thank you for filing this issue with. To set expectations, we’re likely not going to get to it for a while as we’re currently focused on larger coherence work items. However, I see you are willing contribute to this issue, that is great news! 
If you have a solution at hand and would like to contribute towards it, please, share your PR for us and the team will review it!
